pinot-core/src/main/java/com/linkedin/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -84,12 +86,14 @@ public synchronized String getControllerLeader() {
       ZNRecord znRecord = dataAccessor.get("/" + _clusterName + "/CONTROLLER/LEADER", stat, AccessOption.THROW_EXCEPTION_IFNOTEXIST);
       String leader = znRecord.getId();
       int index = leader.lastIndexOf('_');
-      _controllerLeaderHostPort = leader.substring(0, index) + ":" + leader.substring(index + 1);
+      String leaderHost = leader.substring(0, index);
+      int leadePort = Integer.valueOf(leader.substring(index + 1));
+      _controllerLeaderHostPort = new Pair<>(leaderHost, leadePort);
       _refresh = false;
       return _controllerLeaderHostPort;
     } catch (Exception e) {
       LOGGER.warn("Could not locate controller leader, exception", e);
-      _controllerLeaderHostPort = null;
+      _controllerLeaderHostPortNotUsed = null;
```

Comment:
**_controllerLeaderHostPort** = null

Reply:
not used, removed. it was left over from the renaming of the old var